Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -146,32 +176,58 @@ std::map<std::string, std::string> DiscusMultipleScatteringCorrection::validateI
+  if (inputWS->getEMode() == Kernel::DeltaEMode::Elastic) {
+    if (inputWS->getAxis(0)->unit()->unitID() != "Momentum") {
+      issues["InputWorkspace"] = "Input workspace must have units of Momentum (k) for elastic instrument\n";
+    }
+    if (SQWS->getNumberHistograms() != 1) {
+      issues["StructureFactorWorkspace"] = "S(Q) workspace must contain a single spectrum for elastic mode\n";
+    }
+
+    if (SQWS->getAxis(0)->unit()->unitID() != "MomentumTransfer") {
+      issues["StructureFactorWorkspace"] += "S(Q) workspace must have units of MomentumTransfer\n";
+    }
+  } else {
+    if (inputWS->getAxis(0)->unit()->unitID() != "DeltaE") {
+      issues["InputWorkspace"] = "Input workspace must have units of DeltaE for inelastic instrument\n";
+    }
+    std::set<std::string> axisUnits;
+    axisUnits.insert(SQWS->getAxis(0)->unit()->unitID());
+    axisUnits.insert(SQWS->getAxis(1)->unit()->unitID());
+    if (axisUnits != std::set<std::string>{"DeltaE", "MomentumTransfer"}) {
+      issues["StructureFactorWorkspace"] = "S(Q, w) workspace must have units of Energy Transfer and MomentumTransfer";
+    }
   }
+
   auto y = SQWS->y(0);
   if (std::any_of(y.cbegin(), y.cend(), [](const auto yval) { return yval < 0 || std::isnan(yval); })) {
-    issues["SofqWorkspace"] = "S(Q) workspace must have all y >= 0";
-  }
-
-  const std::string emodeStr = getProperty("EMode");
-  auto emode = Kernel::DeltaEMode::fromString(emodeStr);
-  if (emode != Kernel::DeltaEMode::Elastic) {
-    issues["EMode"] = "Only elastic mode is supported at the moment";
+    issues["StructureFactorWorkspace"] += "S(Q) workspace must have all y >= 0";
   }
 
-  const int nlambda = getProperty("NumberOfWavelengthPoints");
-  if (!isEmpty(nlambda)) {
+  const int nSimulationPoints = getProperty("NumberOfSimulationPoints");
+  if (!isEmpty(nSimulationPoints)) {
     InterpolationOption interpOpt;
     const std::string interpValue = getPropertyValue("Interpolation");
     interpOpt.set(interpValue, false, false);
-    const auto nlambdaIssue = interpOpt.validateInputSize(nlambda);
-    if (!nlambdaIssue.empty()) {
-      issues["NumberOfWavelengthPoints"] = nlambdaIssue;
+    const auto nSimPointsIssue = interpOpt.validateInputSize(nSimulationPoints);
+    if (!nSimPointsIssue.empty()) {
+      issues["NumberOfSimulationPoints"] = nSimPointsIssue;
     }
   }
 
+  const bool simulateEnergiesIndependently = getProperty("SimulateEnergiesIndependently");
+  if (simulateEnergiesIndependently) {
+    if (inputWS->getEMode() == Kernel::DeltaEMode::Elastic)
+      issues["SimulateEnergiesIndependently"] =
+          "SimulateEnergiesIndependently is only applicable to inelastic direct geometry calculations";
+    if (inputWS->getEMode() == Kernel::DeltaEMode::Indirect)
+      issues["SimulateEnergiesIndependently"] =
+          "SimulateEnergiesIndependently is only applicable to inelastic direct geometry calculations. Different "
```

Comment:
Perhaps the fact the property `SimulateEnergiesIndependently` applies only to direct geometry could be mentioned in the doc string in  `declareProperty`